clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -285,9 +285,9 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
       if (FVConstraint *FV = dyn_cast<FVConstraint>(F)) {
         // This is to ensure that the return type of the function is same
         // as the type of return expression.
-        constrainConsVarGeq(FV->getReturnVars(), RconsVar,
-                            Info.getConstraints(), &PL, Same_to_Same, false,
-                            &Info);
+        for (ConstraintVariable *CV : RconsVar)
+          constrainConsVarGeq(FV->getReturnVars(), CV, Info.getConstraints(),
+                              &PL, Same_to_Same, false, &Info);
```

Comment:
I don't understand this replacement. How are you moving from a non-loop to a loop if you are *removing* sets?

Reply:
Figured it out.

Reply:
`constrainConsVarGeq` takes either two `std::set<ConstraintVariable *>` or two `ConstraintVariable *`. Since there's no prototype that take a pointer as one argument and a set as the second, I had put this loop here. The `std::set` overloading of the function need two loops, so we do end up with one fewer loops as you would expect. 

I might add a new overloading of the function with this prototype, since it comes in a few places even before the changes I've made.

Reply:
Sure that would work. But what you have is actually fine. Less to look at when just reading the `ConstraintVariables.cpp` file